py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -407,3 +418,44 @@ def get_info(self) -> Tuple[str, Platform, Union[BitBox02Edition, BitBoxBaseEdit
         unlocked_byte = response[0]
         unlocked = {0x00: False, 0x01: True}[unlocked_byte]
         return (version_str, platform, edition, unlocked)
+
+    def firmware_outdated(self, required_version: semver.VersionInfo) -> None:
+        raise VersionOutdatedException(
+            "The BitBox02's firmware is no up to date. Device: {}, Required: {}".format(
+                self.version, required_version
+            )
+        )
+
+    def library_outdated(self) -> None:
+        raise VersionOutdatedException(
+            "The BitBox02's firmware version {} is too new for this app. Update the app".format(
+                self.version
+            )
+        )
+
+    def check_version(self) -> None:
+        """
+        Returns false if the device has an older firmware version than required
+        and the minimum required version.
+        """
+        _, platform, edition, _ = self.get_info()
+        if platform == Platform.BITBOX02:
+            if edition == BitBox02Edition.MULTI:
+                if self.version < MIN_BITBOX02_MULTI_FIRMWARE_VERSION:
+                    self.firmware_outdated(MIN_BITBOX02_MULTI_FIRMWARE_VERSION)
```

Comment:
nit: `raise self.firmware_outdated(...)` would be a bit easier to read, as like this it's not clear that this stops the function.

Reply:
Done, moved everything into separate exception classes.